File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.js
const { Console } = require('console')
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split('\n')

input = input.map(x => parseInt(x))

function check(start, end){
    let arr = new Array(end+1).fill(true)
    
    for (let i = 2; i < parseInt(end ** 0.5)+1; i++) {
        if (arr[i]) {
            for (let j = 2*i; j < end+1; j+=i) {
                arr[j] = false
            }
        }
    }

    let cnt = 0
    for (k of arr.slice(start+1, end+1)) {
        if (k) {
            cnt += 1
        }
    }
    return cnt 
}

input.forEach(x => {if (x !== 0) {console.log(check(x, 2*x))}})

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
import sys, math
input = sys.stdin.readline


def prime_list(start, n):

    arr = [True] * (n+1)
    m = int(n ** 0.5)

    for i in range(2, m+1):
        if arr[i] == True:
            for j in range(2*i, n+1, i):
                arr[j] = False

    arr = [i for i in (arr[start+1:n+1]) if i == True]

    return len(arr)

while True:
    x = int(input())

    if x == 0:
        exit(0)

    print(prime_list(x,2*x))